File: Graph Algorithms in Genome Sequencing/Genome seq/reconstruct string from its Genome path/StringFromGenomePath.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.List;

public class StringFromGenomePath {
    /*
     * Reconstructs a string from its genome path
     *
     * @param patterns      a list of String k-mers in the order they appear
     *                      in text
     *
     * @return a String text with the given genome path
     */
    public static String reconstructString(List<String> patterns) {
        // TODO: your code here
        return "";
    }

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        List<String> patterns = new ArrayList<String>();

        while (stdin.hasNextLine()) patterns.add(stdin.nextLine());

        System.out.println(reconstructString(patterns));
    }
}

File: Graph Algorithms in Genome Sequencing/Genome seq/reconstruct string from its Genome path/string_from_genome_path.cpp
#include <iostream>
#include <vector>
#include <string>

using std::cout;
using std::cin;
using std::endl;
using std::string;
using std::vector;


/*
 * Reconstructs a string from its genome path
 *
 * \param patterns      a vector of string k-mers in the order they appear
 *                      in text
 *
 * \return a string text with the given genome path
 */
string reconstruct_string(const vector<string>& patterns) {
    // TODO: your code here
    return "";
}


int main(int argc, char* argv[]) {
    vector<string> patterns;

    string pattern;
    while (cin >> pattern) {
        patterns.push_back(pattern);
    }

    cout << reconstruct_string(patterns) << endl;
}
